paserk/paserk_test.go

```diff
@@ -0,0 +1,45 @@
+package paserk
```

Comment:
My only comment is it might be better to run the [official test suite here](https://github.com/paragonie/paserk-php/blob/master/tests/test-vectors/k3.secret.json).

You could use `go generate` to embed all the test vectors into the repo and run the tests.

Reply:
I could copy the JSON file into the repo like it's done for the PASETO test vectors and run those additionally

Reply:
Added them now.
Some of the test cases make no sense imo, like `k2.secret-fail-2` which doesn't test PASERK itself. Instead, it seems to be meant to assert that a `V4SymmetricKey` can't be serialized as e.g. `k1.local`. Since the target type is inferred from the key itself, and it is tested whether a key serializes to the correct PASERK type by the other tests, I think it's fine that the test just does nothing (it hits the expected fail branch since the PASERK in the test data is empty). Same applies for the other expected-fail tests that are meant to test if a cut short key will serialize.